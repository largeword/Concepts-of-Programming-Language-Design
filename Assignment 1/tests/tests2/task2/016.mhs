main :: [Int]
    = let list :: (Int -> [Int] -> [Int])
        = recfun list :: (Int -> [Int] -> [Int]) = Cons;
    in list 12 Nil; -- returns Cons 12 Nil
