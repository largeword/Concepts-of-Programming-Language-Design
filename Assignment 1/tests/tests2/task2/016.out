(Cons 12 Nil)